### PR TITLE
1270 primitive array consistency

### PR DESCRIPTION
**Pull request by masterav10**
#### Check List:
* Prerequisite to #1270
* Unit tests : YES
* Javadoc with a code example (on API only) : NA

While working on 1270, I noticed some inconsistencies in the array implementations. To prevent change overload, here are my proposed changes to the abstract array base classes in support for 1270.

Changes so far:
- [X] AbstractByteArrayAssert.containsExactly.containsExactly() should use arrays internally
- [X] AbstractShortArrayAssert added int methods for like the byte array version

Potential other options I wasn't planning on doing:
- AbstractCharArrayAssert add int methods
- AbstractLongArrayAssert add int methods
- AbstractFloatArrayAssert add double methods

### CODE REVIEW

File: src/test/java/org/assertj/core/api/shortarray/ShortArrayAssertNullTest.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * Should be used for methods with an int[] or int... parameter.
+ * 
+ * @author Dan Avila
+ */
+public abstract class ShortArrayAssertNullTest extends ShortArrayAssertBaseTest {
+
+  @Test
+  public void should_throw_exception_on_null_argument() {
+    assertThatExceptionOfType(NullPointerException.class).isThrownBy(() -> {
+      int[] content = null;
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/shortarray/ShortArrayAssertNullTest.java]: Let's use `catchThrowableOfType` and the GIVEN WHEN THEN pattern to make the test easier to read.  
shoud look like:
```java
    // GIVEN
    int[] nullContent = null;
    // WHEN
    NullPointerException npe = catchThrowableOfType(() -> invoke_api_with_null_value(assertThat(new short[] {}), nullContent),
                                                    NullPointerException.class);
    // THEN
    then(npe).hasMessage("The array of values to look for should not be null");
```

File: src/test/java/org/assertj/core/api/shortarray/ShortArrayAssert_containsAnyOf_with_Integer_Argument_Test.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import org.assertj.core.api.AbstractShortArrayAssert;
+import org.assertj.core.api.ShortArrayAssert;
+import org.junit.jupiter.api.DisplayName;
+
+/**
+ * Tests for <code>{@link ShortArrayAssert#containsAnyOf(int...)}</code>.
+ * 
+ * @author Dan Avila
+ */
+@DisplayName("ShortArrayAssert containsAnyOf (ints)")
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/shortarray/ShortArrayAssert_containsAnyOf_with_Integer_Argument_Test.java]: nice!

Comment (masterav10) [reply src/test/java/org/assertj/core/api/shortarray/ShortArrayAssert_containsAnyOf_with_Integer_Argument_Test.java]: Glad you liked it.

**Comment by masterav10** [comment]
Should be good to go. I've made pretty good progress on the parent branch as well.

**Comment by joel-costigliola** [comment]
@masterav10 I'm not sure what you mean by the parent branch but this PR looks good so I'm integrating it.

**Pull request closed without merging**